resources/deb/etc/init.d/openhab2

```diff
@@ -71,15 +71,36 @@ fi
 # Load the VERBOSE setting and other rcS variables
 . /lib/init/vars.sh
 
+# Check to see if openHAB is currently running
+EXISTINGPID=`pgrep java -G openhab | tail -1`
+
 case "$1" in
   start)
         log_daemon_msg "Starting $DESC" "$NAME"
-        #${OPENHAB_INST_DIR}/bin/setpermissions.sh > /dev/null
-        if start-stop-daemon --start --quiet --pidfile $PIDFILE -c "${USER_AND_GROUP}" --umask 002 --background --make-pidfile --chdir ${OPENHAB_DIR} \
-                --oknodo --exec /usr/share/openhab2/runtime/bin/start
+
+        # If a process still exists but there's no PID file, then openHAB is probably still shutting down. We should proceed.
+        if [ -z "$EXISTINGPID" ] || [ ! -f $PIDFILE ];
         then
-            log_end_msg 0
+            if start-stop-daemon --start --quiet --pidfile $PIDFILE -c "${USER_AND_GROUP}" --umask 002 --background --chdir ${OPENHAB_DIR} \
+                    --oknodo --exec /usr/share/openhab2/runtime/bin/start
+            then
+                timeout=0
+                # The correct process is only available once the wrapper scripts have finished, we should prevent the user from running again
+                until [ $timeout -eq 5 ] || [ ! -z "$pid"  ];
+                do
+                    sleep 1
+                    pid=`pgrep java -G openhab | tail -1`
+                    timeout=$((timeout+1))
+                done
+                echo "$pid" > $PIDFILE
+                log_end_msg 0
+            else
+                log_end_msg 1
+            fi
         else
+            echo -n "Error: OpenHAB is already running with PID: $EXISTINGPID!"
+            # Update PID File incase it is stale.
+            echo "$EXISTINGPID" > $PIDFILE
```

Comment:
IMHO the init script shouldn't try to fix an inconsistent system state. It should be kept as small and simple as possible and focus on the task of (trying) to start/stop the service and get the status of the service. The script may assume that the service is only handled by the init script itself.
If an admin has started openHAB without using the init script he should be able to fix it on his own.